File: javascript/station-capabilities.js
// Which normals datasets exist for a given station.
const BASE_URL = "https://api.usclimatedata.com/v1";
const API_KEY = process.env.USCLIMATEDATA_API_KEY;
const STATION_ID = "USFL0316";

const response = await fetch(`${BASE_URL}/stations/${STATION_ID}/capabilities`, {
  headers: { "X-API-Key": API_KEY },
});
if (!response.ok) {
  throw new Error(`Request failed: ${response.status}`);
}
console.log(await response.json());

File: python/station_capabilities.py
"""Which normals datasets exist for a given station."""
import os

import requests

BASE_URL = "https://api.usclimatedata.com/v1"
API_KEY = os.environ["USCLIMATEDATA_API_KEY"]
STATION_ID = "USFL0316"

response = requests.get(
    f"{BASE_URL}/stations/{STATION_ID}/capabilities",
    headers={"X-API-Key": API_KEY},
    timeout=10,
)
response.raise_for_status()
print(response.json())
